diff --git a/src/codeeditor/codeeditor.cpp b/src/codeeditor/codeeditor.cpp
--- a/src/codeeditor/codeeditor.cpp
+++ b/src/codeeditor/codeeditor.cpp
@@ -568,6 +568,6 @@ se::GestureEvent ToWheelGestureEvent(const CodeEditorScrollEvent& event) {
   se::GestureEvent out;
   out.type = se::EventType::MOUSE_WHEEL;
-  out.wheel_delta_x = event.delta_x;
-  out.wheel_delta_y = event.delta_y;
+  out.wheel_delta_x = -event.delta_x;
+  out.wheel_delta_y = -event.delta_y;
   return out;
 }
